src/integrations/watchers/auto-update-watcher.php

```diff
@@ -0,0 +1,215 @@
+		$this->notification_helper = $notification_helper;
+		$this->product_helper      = $product_helper;
+	}
+
+	/**
+	 * Initializes the integration.
+	 *
+	 * On admin_init, it is checked whether the notification to auto-update Yoast SEO needs to be shown or removed.
+	 * This is also done when major WP core updates are being enabled or disabled,
+	 * and when automatic updates for Yoast SEO are being enabled or disabled.
+	 *
+	 * @return void
+	 */
+	public function register_hooks() {
+		\add_action( 'admin_init', [ $this, 'auto_update_notification_not_if_dismissed' ] );
+		\add_action( 'update_option_auto_update_core_major', [ $this, 'auto_update_notification_even_if_dismissed' ] );
+		\add_action( 'update_option_auto_update_plugins', [ $this, 'auto_update_notification_not_if_dismissed' ] );
+	}
+
+	/**
+	 * Handles the Yoast SEO auto-update notification when the user toggles the auto-update setting for WordPress Core.
+	 *
+	 * If it should be shown, this will be done even if the notification has been dismissed in the past.
+	 *
+	 * @return void
+	 */
+	public function auto_update_notification_even_if_dismissed() {
+		if ( ! $this->should_show_notification() ) {
+			$this->save_dismissal_status();
+			$this->maybe_remove_notification();
+
+			return;
+		}
+
+		$this->maybe_create_notification();
+	}
+
+	/**
+	 * Handles the Yoast SEO auto-update notification on all admin pages,
+	 * as well as when the user toggles the Yoast SEO auto-update setting.
+	 *
+	 * If it should be shown, this will only be done if the notification has not been dismissed in the past.
+	 *
+	 * @return void
+	 */
+	public function auto_update_notification_not_if_dismissed() {
+		if ( ! $this->should_show_notification() ) {
+			$this->save_dismissal_status();
+			$this->maybe_remove_notification();
+
+			return;
+		}
+
+		$this->maybe_create_notification_if_not_dismissed();
+	}
+
+	/**
+	 * Checks whether the Yoast SEO auto-update notification should be shown.
+	 *
+	 * @return bool Whether the notification should be shown.
+	 */
+	protected function should_show_notification() {
+		$core_updates_enabled  = \get_option( 'auto_update_core_major' ) === 'enabled';
+		$yoast_updates_enabled = $this->yoast_auto_updates_enabled();
+
+		return $core_updates_enabled && ! $yoast_updates_enabled;
+	}
+
+	/**
+	 * Saves the dismissal status of the notification in an option in wp_usermeta, if the notification gets dismissed.
+	 *
+	 * @return void
+	 */
+	protected function save_dismissal_status() {
+		// This option exists if the notification has been dismissed at some point.
+		$notification_dismissed = \get_user_option( 'wp_' . self::NOTIFICATION_ID );
+
+		// We wish to have its value in a different option, so we can still access it even when the notification gets removed.
+		if ( $notification_dismissed && ! \get_user_option( 'wp_' . self::NOTIFICATION_ID . '_dismissed' ) ) {
+			\update_user_option( \get_current_user_id(), self::NOTIFICATION_ID . '_dismissed', true );
```

Comment:
If you decide to keep the `wp_` prefix, I think this needs a prefix as wel.

Reply:
Apparently not, it appears in the database as `wp_wpseo-auto-update_dismissed`. So I think under the hood WordPress somewhere adds the `wp_` prefix.